articles/user-guide/host-programs.md

```diff
@@ -0,0 +1,455 @@
+
+            Console.WriteLine($"Single qubit result: {single_qubit_result}");
+            Console.WriteLine($"Multiple qubits result: {multi_qubit_result}");
+        }
+    }
+}
+```
+
+At the location of the C# file, the host program can be run from the command line by entering
+
+```bash
+dotnet run
+```
+
+and in this case you will see output written to the console similar to 
+
+```bash
+Single qubit result: Zero
+Multiple qubits result: [One,One,Zero,Zero]
+```
+
+> [!NOTE]
+> Due to the compiler's interoperability with namespaces, we could alternatively make our Q# operations available without the `using NamespaceName;` statement, and simply matching the C# namespace title to it.
+> That is, replacing `namespace host` with `namespace NamespaceName`.
+
+***
+
+## Q# Jupyter Notebooks
+Q# Jupyter Notebooks make use of the IQ# kernel, which allows you to define, compile, and run Q# operations in a single notebook---all alongside instructions, notes, and other content.
+This means that while it is possible to import and use the contents of `*.qs` Q# files, they are not necessary in the execution model.
+
+Here, we will detail how to run the Q# operations defined above, but another good introduction to using Q# Jupyter Notebooks is provided at [Intro to Q# and Jupyter Notebook](https://github.com/microsoft/Quantum/blob/master/samples/getting-started/intro-to-iqsharp/Notebook.ipynb).
+
+### Defining operations
+
+In a Q# Jupyter Notebook, you enter Q# code just as we would from inside the namespace of a Q# file.
+
+So, we can enable access to operations from the [Q# standard libraries](xref:microsoft.quantum.qsharplibintro) with `open` statements for their respective namespaces.
+Upon running a cell with such a statement, the definitions from those namespaces are available throughout the workspace.
+
+> [!NOTE]
+> Operations from [Microsoft.Quantum.Intrinsic](xref:microsoft.quantum.intrinsic) and [Microsoft.Quantum.Canon](xref:microsoft.quantum.canon) (e.g. [`H`](xref:microsoft.quantum.intrinsic.h) and [`ApplyToEach`](xref:microsoft.quantum.canon.applytoeach)) are automatically available in Q# Jupyter Notebooks.
+
+Similarly, defining operations requires only writing the Q# code and running the cell.
+
+<img src="./images/jupyter_op_def_crop.png" alt="Jupyter cell defining Q# operations" width="600">
+
+The output then lists those operations, which can then be called from future cells.
+
+### Target machines
+
+The functionality to run operations on specific target machines is provided via [IQ# Magic Commands](xref:microsoft.quantum.guide.quickref.iqsharp).
+For example, `%simulate` makes use of the `QuantumSimulator`, and `%estimate` uses the `ResourcesEstimator`:
+
+<img src="./images/jupyter_no_args_sim_est_crop.png" alt="Simulate and estimate resources Jupyter cell" width="500">
+
+### Arguments
+
+Currently the execution magic commands can only be used with operations that take no arguments. 
```

Comment:
+@rmshaffer for awareness.

Reply:
Thanks! I've opened a task for myself to update this once passing arguments is allowed.

Reply:
Two quick updates here: As of the next release, the execution magic commands will be able to accept arguments. And also, the syntax highlighting for magic commands has been updated, so probably a couple of the screenshots here will need to be replaced. I'm planning to make those updates here; this is tracked by AB#15207.